File: src/main/java/ru/pearx/pmdumper/dumpers/DumperEnchantments.java
package ru.pearx.pmdumper.dumpers;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.ForgeRegistries;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import ru.pearx.pmdumper.dumpers.base.IDumper;
import ru.pearx.pmdumper.dumpers.base.PMDData;

import java.util.*;

/**
 * Created by mrAppleXZ on 12.06.17 12:20.
 */
@SideOnly(Side.CLIENT)
public class DumperEnchantments implements IDumper
{

    @Override
    public String getName()
    {
        return "enchantments";
    }

    @Override
    public PMDData getData()
    {
        List<List<String>> lst = new ArrayList<>();
        Map<String, Integer> counts = new HashMap<>();
        lst.add(Arrays.asList("ID", "Name", "Class Name", "Levels", "Rarity", "Is Curse", "Type", "Allowed on books", "Is Treasure", "Level (Min Enchantability - Max Enchantability)..."));
        for(Map.Entry<ResourceLocation, Enchantment> entr : ForgeRegistries.ENCHANTMENTS.getEntries())
        {
            ResourceLocation loc = entr.getKey();
            Enchantment ench = entr.getValue();
            List<String> row = new ArrayList<>();
            row.add(loc.toString());
            row.add(ench.getName());
            row.add(ench.getClass().getName());
            row.add(ench.getMinLevel() + " - " + ench.getMaxLevel());
            row.add(ench.getRarity().toString());
            row.add(Boolean.toString(ench.func_190936_d()));
            row.add(ench.type == null ? "null" : ench.type.toString());
            row.add(Boolean.toString(ench.isAllowedOnBooks()));
            row.add(Boolean.toString(ench.isTreasureEnchantment()));
            for(int i = ench.getMinLevel(); i <= ench.getMaxLevel(); i++)
            {
                row.add(ench.getTranslatedName(i) + " (" + ench.getMinEnchantability(i) + " - " + ench.getMaxEnchantability(i) + ")");
            }
            PMDData.plusCounts(counts, loc.getResourceDomain(), 1);
            lst.add(row);
        }
        return new PMDData(lst, counts);
    }
}

File: src/main/kotlin/ru/pearx/pmdumper/dumper/standard/DumperEnchantments.kt
@file:JvmMultifileClass
@file:JvmName("StandardDumpers")
package ru.pearx.pmdumper.dumper.standard

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.common.registry.ForgeRegistries
import ru.pearx.pmdumper.ID
import ru.pearx.pmdumper.dumper.dumper
import ru.pearx.pmdumper.utils.toPlusMinusString

val DumperEnchantments = dumper {
    registryName = ResourceLocation(ID, "enchantments")
    header = listOf("ID", "Name", "Class Name", "Levels", "Rarity", "Is Curse", "Type", "Allowed on Books", "Is Treasure", "Localized Name [Min Enchantability - Max Enchantability]")
    amounts {
        for (ench in ForgeRegistries.ENCHANTMENTS)
            this += ench.registryName
    }
    iterator {
        for (ench in ForgeRegistries.ENCHANTMENTS) {
            with(ArrayList<String>(header.size)) {
                with(ench) {
                    add(registryName.toString())
                    add(name)
                    add(this::class.java.name)
                    add("$minLevel - $maxLevel")
                    add(rarity.toString())
                    add(isCurse.toPlusMinusString())
                    add(type.toString())
                    add(isAllowedOnBooks.toPlusMinusString())
                    add(isTreasureEnchantment.toPlusMinusString())
                    add(StringBuilder().apply {
                        var start = true
                        for (lvl in minLevel..maxLevel) {
                            if (start)
                                start = false
                            else
                                appendln()
                            append("${getTranslatedName(lvl)} [${getMinEnchantability(lvl)} - ${getMaxEnchantability(lvl)}]")
                        }
                    }.toString())
                }
                yield(this)
            }
        }
    }
}